noether_filtering/src/mesh/windowed_sinc_smoothing.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * window_sinc_smoothing.cpp
+ *
+ *  Created on: Dec 6, 2019
+ *      Author: jrgnicho
+ */
+
+#include <XmlRpcException.h>
+#include <vtkSmartPointer.h>
+#include <vtkPolyData.h>
+#include <vtkWindowedSincPolyDataFilter.h>
+#include <pcl/surface/vtk_smoothing/vtk_utils.h>
+#include <console_bridge/console.h>
+#include "noether_filtering/utils.h"
+#include "noether_filtering/mesh/windowed_sinc_smoothing.h"
+
+namespace noether_filtering
+{
+namespace mesh
+{
+WindowedSincSmoothing::WindowedSincSmoothing()
+{
+
+}
+
+WindowedSincSmoothing::~WindowedSincSmoothing()
+{
+
+}
+
+bool WindowedSincSmoothing::configure(XmlRpc::XmlRpcValue config)
+{
+  std::vector<std::string> fields = {"num_iter",
+                                     "enable_boundary_smoothing",
+                                     "enable_feature_edge_smoothing",
+                                     "enable_non_manifold_smoothing",
+                                     "enable_normalize_coordinates",
+                                     "feature_angle",
+                                     "edge_angle",
+                                     "pass_band",
+                                     };
+
+  if(!std::all_of(fields.begin(), fields.end(), [&config, this](const std::string& f){
+    if( !config.hasMember(f))
+    {
+      CONSOLE_BRIDGE_logError("The %s config field %s was not found", getName().c_str(), f.c_str());
+      return false;
+    }
+    return true;
+  }
+  ))
+  {
+    return false;
+  }
+
+  try
+  {
+    std::size_t idx = 0;
+    config_.num_iter = static_cast<int>(config[fields[idx++]]);
```

Comment:
This seems error prone. Success depends on the above initialization of fields matches the order. Seems like you could use a map, still check to insure the config has each member, and then select the right config value.

Reply:
Yes the order of assignment into the `config` structure needs to match the order the field names are listed in the `fields` vector.  The yaml however can have the fields in any order.  I'm not sure that using a map would be more advantageous plus it would required setting up an `enum` with entries corresponding to each field  that can then be used as the keys in the map